src/backend/drm/compositor/mod.rs

```diff
@@ -2930,6 +2946,26 @@ where
         Ok(())
     }
 
+    /// Returns if Variable Refresh Rate is advertised as supported by the given connector.
+    ///
+    /// See [`DrmSurface::vrr_supported`] for more details.
+    pub fn vrr_supported(&self, conn: connector::Handle) -> FrameResult<VrrSupport, A, F> {
```

Comment:
Maybe we can add a `vrr_enabled` function, returning the current vrr state. Could be useful when sending presentation feedback.

Reply:
I would expect reading back the property on every frame to be quite slow and instead expect compositors to track the state themselves. Is that a wrong assumption?

Reply:
> But that actually made me question how we would handle wp-presentation v2, which requires us to send the minimum refresh.

Oh right, good idea.

> But that actually made me question how we would handle wp-presentation v2, which requires us to send the minimum refresh. So maybe it already has to be tracked somewhere else?

Maybe we should just merge #1588, so I can more easily experiment with this. But it might make sense to add this to the `RenderFrameResult`. Compositors would then need to pass this through the `queue_frame` user-data to have access to this after the vblank-event.

Reply:
Yeah, sounds good. I will it merge it now :)

Reply:
The min refresh rate is defined in the monitor EDID iirc. libdisplayinfo should already expose it.

Reply:
> For version 2 and later, if the output does not have a constant refresh rate, explicit video mode switches excluded, then the refresh argument must be either an appropriate rate picked by the compositor (e.g. fastest rate), or 0 if no such rate exists.

Doesn't that mean the value should simply be that of the `Mode`?